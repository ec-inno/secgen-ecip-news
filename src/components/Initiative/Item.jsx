import React, { useState, useEffect } from 'react';
import axios from 'axios';

import formatStatus from '../../utils/formatStatus';
import getCurrentLanguage from '../../utils/getCurrentLanguage';
import getDefaultLanguage from '../../utils/getDefaultLanguage';

import defaultImage from '../assets/images/default-image.png';

const InitiativeItem = ({ item, location }) => {
  const language = getCurrentLanguage(location) || getDefaultLanguage();
  const translation = require(`../../../translations/initiative/${language}.json`);

  const { GATSBY_INITIATIVES_API: api } = process.env;

  const [logo, setLogo] = useState('');

  useEffect(() => {
    if (item.logo && item.logo.id) {
      axios
        .get(`${api}/register/logo/${item.logo.id}`, {
          responseType: 'arraybuffer',
        })
        .then(response => {
          const logoData = Buffer.from(response.data, 'binary').toString(
            'base64'
          );
          setLogo(logoData);
        })
        .catch(e => {
          console.error(`Failed to fetch logo ${item.logo.id}`, e);
        });
    }
  }, []);

  const background = logo
    ? `data:${item.logo.mimeType};base64,${logo}`
    : defaultImage;

  return (
    <article className="ecl-card">
      <header className="ecl-card__header">
        <a href="#" className="ecl-tag ecl-u-f-r eci-tag">
          {formatStatus(item.status)}
        </a>
        <div
          className="ecl-card__image"
          alt={`Image of ${item.title} initiative`}
          style={{ backgroundImage: `url('${background}')` }}
        ></div>
        <h1 className="ecl-card__title">
          <a
            level="1"
            href={`/${language}/initiatives/#${item.id}`}
            className="ecl-link ecl-link--standalone"
          >
            <span className="ecl-link__label">{item.title}</span>
          </a>
        </h1>
      </header>
      <section className="ecl-card__body">
        <p className="ecl-u-type-paragraph ecl-u-mv-none">
          {item.totalSupporters} {translation.supporters}
        </p>
        {item.supportLink && (
          <ul className="ecl-card__link-container">
            <li className="ecl-card__link-item">
              <a
                target="_blank"
                href={item.supportLink}
                className="ecl-card__link ecl-link ecl-link--standalone"
              >
                {translation.support_cat}
              </a>
            </li>
          </ul>
        )}
      </section>
    </article>
  );
};

export default InitiativeItem;
