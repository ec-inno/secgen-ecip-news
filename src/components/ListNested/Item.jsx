import React from 'react';
import PropTypes from 'prop-types';
import { Link as LinkInternal } from 'gatsby';
import LinkExternal from '../Link/LinkEcl';

const Item = ({ item, id }) => {
  const accepted = ['href', 'title', 'external'];

  if (!item) return '';

  let ul = null;
  const children = {};

  Object.keys(item).forEach(prop => {
    if (!accepted.includes(prop)) {
      children[prop] = item[prop];
    }
  });

  if (Object.keys(children).length) {
    ul = (
      <ul className="ecl-unordered-list">
        {Object.keys(children).map((childPath, key) => (
          <Item item={children[childPath]} key={key} />
        ))}
      </ul>
    );
  }

  return (
    <li className="ecl-unordered-list__item" key={id}>
      {item.external ? (
        <LinkExternal
          href={item.href}
          className="ecl-link ecl-link--standalone"
        >
          {item.title}
        </LinkExternal>
      ) : (
        <LinkInternal to={item.href} className="ecl-link ecl-link--standalone">
          {item.title}
        </LinkInternal>
      )}
      {ul}
    </li>
  );
};

Item.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string,
    href: PropTypes.string,
    external: PropTypes.bool,
  }),
  id: PropTypes.string,
};

export default Item;
