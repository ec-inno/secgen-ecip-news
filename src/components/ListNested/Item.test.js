import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';

describe('Item', () => {
  it('Returns an empty string on missing input', () => {
    const tree = renderer.create(<Item />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
