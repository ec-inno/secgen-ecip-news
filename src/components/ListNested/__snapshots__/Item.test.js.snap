// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Returns an empty string on missing input 1`] = `""`;
